js/graphql.md

```diff
@@ -0,0 +1,1346 @@
+  id: ID!
+  title: String!
+  owners: [String]
+}
+```
+
+**Static Group Authorization**
+
+```
+# Static group auth
+type Post @model @auth(rules: [{allow: groups, groups: ["Admin"]}]) {
+  id: ID!
+  title: String
+}
+```
+
+If the user credential (as specified by the resolver's `$ctx.identity`) is not
+enrolled in the *Admin* group, throw an unauthorized error using `$util.unauthorized()`.
+
+**Dynamic Group Auth**
+
+```
+# Dynamic group authorization with multiple groups
+type Post @model @auth(rules: [{allow: groups, groupsField: "groups"}]) {
+  id: ID!
+  title: String
+  groups: [String]
+}
+
+# Dynamic group authorization with a single group
+type Post @model @auth(rules: [{allow: groups, groupsField: "group"}]) {
+  id: ID!
+  title: String
+  group: String
+}
+```
+
+With dynamic group authorization, each record contains an attribute specifying
+what groups should be able to access it. Use the *groupsField* argument to
+specify which attribute in the underlying data store holds this group
+information. To specify that a single group should have access, use a field of
+type `String`. To specify that multiple groups should have access, use a field of
+type `[String]`.
+
+#### Generates
+
+The `@auth` directive will add authorization snippets to any relevant resolver 
+mapping templates at compile time. Different operations use different methods
+of authorization.
+
+**Owner Authorization**
+
+```
+type Post @model @auth(rules: [{allow: owner}]) {
+  id: ID!
+  title: String!
+}
+```
+
+the generated resolvers would be protected like so:
+
+- `Mutation.createX`: Verify the requesting user has a valid credential and automatically set the **owner** attribute to equal `$ctx.identity.username`.
+- `Mutation.updateX`: Update the condition expression so that the DynamoDB `UpdateItem` operation only succeeds if the record's **owner** attribute equals the caller's `$ctx.identity.username`.
+- `Mutation.deleteX`: Update the condition expression so that the DynamoDB `DeleteItem` operation only succeeds if the record's **owner** attribute equals the caller's `$ctx.identity.username`.
+- `Query.getX`: In the response mapping template verify that the result's **owner** attribute is the same as the `$ctx.identity.username`. If it is not return null.
+- `Query.listX`: In the response mapping template filter the result's **items** such that only items with an **owner** attribute that is the same as the `$ctx.identity.username` are returned.
+
+**Multie Owner Authorization**
+
+Work in progress.
+
+**Static Group Authorization**
+
+```
+type Post @model @auth(rules: [{allow: groups, groups: ["Admin"]}]) {
+  id: ID!
+  title: String!
+  groups: String 
+}
+```
+
+Static group auth is simpler than the others. The generated resolvers would be protected like so:
+
+- `Mutation.createX`: Verify the requesting user has a valid credential and that `ctx.identity.claims.get("cognito:groups")` contains the **Admin** group. If it does not, fail`.
```

Comment:
You have a trailing backtick here that I think is messing up the rest of the markdown.

Reply:
+1